src/cursor/cur_backup.c

```diff
@@ -299,12 +301,15 @@ __wt_curbackup_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *other,
         F_SET(cb, WT_CURBACKUP_QUERYID);
     }
 
+    /* Special backup cursor for export operation. */
+    export_backup = (strcmp(uri, "backup:export") == 0) ? true : false;
+
     /*
      * Start the backup and fill in the cursor's list. Acquire the schema lock, we need a consistent
      * view when creating a copy.
      */
-    WT_WITH_CHECKPOINT_LOCK(
-      session, WT_WITH_SCHEMA_LOCK(session, ret = __backup_start(session, cb, othercb, cfg)));
+    WT_WITH_CHECKPOINT_LOCK(session,
+      WT_WITH_SCHEMA_LOCK(session, ret = __backup_start(session, cb, othercb, cfg, export_backup)));
```

Comment:
Instead of adding another argument to `backup_start` will it work to model it like the `backup:query_id` code in the same spot and add a flag to the backup cursor `WT_CURBACKUP_EXPORT`. That cursor (and therefore that flag) is passed everywhere in the code and you'd have that information always available.

Reply:
Yes, that will definitely work. I was using the flag approach initially instead of the boolean - I wasn't sure if the flag will be useful if we are only using it in one section of the code but I can definitely change it back to flag.